components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/config/OAuthServerConfiguration.java

```diff
@@ -666,6 +668,34 @@ public boolean isRefreshTokenRenewalEnabled() {
         return supportedResponseTypes;
     }
 
+    public Map<String, ResponseTypeHandler> getDefaultSupportedResponseTypes() {
+        if (defaultSupportedResponseTypes == null) {
+            synchronized (this) {
+                if (defaultSupportedResponseTypes == null) {
```

Comment:
you have initialized this value with an empty map. So, is there any possibility of this being null? (Did you intend an empty check instead? )